Core/include/Acts/Geometry/TrackingGeometryVisitor.hpp

```diff
@@ -92,4 +95,119 @@ class TrackingGeometryMutableVisitor {
   virtual void visitBoundarySurface(BoundarySurfaceT<TrackingVolume>& boundary);
 };
 
+namespace detail {
+
+template <typename Callable>
+consteval bool callableWithAnyMutable() {
+  return std::is_invocable_v<Callable, Surface&> ||
+         std::is_invocable_v<Callable, TrackingVolume&> ||
+         std::is_invocable_v<Callable, BoundarySurfaceT<TrackingVolume>&> ||
+         std::is_invocable_v<Callable, Layer&> ||
+         std::is_invocable_v<Callable, Portal&>;
+}
```

Comment:
`concept` does not work here because of the template argument?

Reply:
I guess a concept would work, yeah. Do you prefer that?

Reply:
no need to change. I just wondered if there is an advantage over concepts